### PR TITLE
♻️ 💄 Refactor Namespaces and API interfaces

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Split the main `IConsumerApi` interface into smaller, UseCase-specific interfaces.
    - i.e.: `IUserTaskConsumerApi`, `IManualTaskConsumerApi`, `IEventConsumerApi`, asf.
    - These interfaces lie in a subfolder called `apis`
    - The main `IConsumerApi` interface no longer has any logic of its own, but instead combines all the UseCase API interfaces.
    - This means that `IConsumerApi` can be used exactly like before.
2. Move the `messages` folder out of the `data_models` folder and place it at root level.
3. Move all API-interfaces, DataModels and Messages into namespaces of the same name.
    - A `Message` namespace already existed, so no changes there.
    - API interfaces can be accessed through the Namespace `APIs.IUserTaskConsumerApi`, asf.
    - The namespaces for DataModels are equal to the folder structure in which the models lie.
    - For example, the type stored under `src/data_models/user_tasks/user_task_list.ts` can be found in the namespace `DataModels.UserTasks.UserTaskList`, asf.
    - `IConsumerApi`, `IConsumerApiHttpController` and `IConsumerApiAccessor` are still located at root level, so no namespace adjustment is required.
4. Use `~` type dependencies for all packages.

PR: #52

## How can others test the changes?

The interface `IConsumerApi` can be accessed like before. Same goes for messages.
When accessing data types, you must now use the `DataModels` namespace.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: typescript/src/apis/ievent_consumer_api.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+   * Retrieves a list of all triggerable events belonging to a correlation.
+   *
+   * @async
+   * @param identity      The requesting users identity.
+   * @param correlationId The ID of the Correlation for which to retrieve
+   *                      the events.
+   * @returns             A Promise, which resolves with the retrieved events,
+   *                      or rejects an error, in case the request failed.
+   *                      This can happen, if the ProcessModel was not found,
+   *                      or the user is not authorized to see the it.
```
Comment (SteffenKn) [inline typescript/src/apis/ievent_consumer_api.ts]:
```suggestion
   *                      or the user is not authorized to see it.
```

File: typescript/src/apis/ievent_consumer_api.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+   * @param identity       The requesting users identity.
+   * @param correlationId  The ID of the Correlation for which to retrieve
+   *                       the events.
+   * @param processModelId The ID of the ProcessModel for which to retrieve
+   *                       the events.
+   * @returns              A Promise, which resolves with the retrieved events,
+   *                       or rejects an error, in case the request failed.
+   *                       This can happen, if the ProcessModel or Correlation
+   *                       was not found, or the user is not authorized to see
+   *                       the it.
```
Comment (SteffenKn) [inline typescript/src/apis/ievent_consumer_api.ts]:
```suggestion
   *                       it.
```

File: typescript/src/apis/imanual_task_consumer_api.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+   * specific ProcessModel.
+   *
+   * @async
+   * @param identity       The requesting users identity.
+   * @param processModelId The ID of the ProcessModel for which to retrieve the
+   *                       ManualTasks.
+   * @returns              A Promise, which resolves with the retrieved ManualTasks,
+   *                       or rejects an error, in case the request failed.
+   *                       This can happen, if the ProcessModel was not found,
+   *                       or the user is not authorized to see the it.
```
Comment (SteffenKn) [inline typescript/src/apis/imanual_task_consumer_api.ts]:
```suggestion
   *                       or the user is not authorized to see it.
```

File: typescript/src/apis/imanual_task_consumer_api.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+   * Correlation.
+   *
+   * @async
+   * @param identity      The requesting users identity.
+   * @param correlationId The ID of the Correlation for which to retrieve the
+   *                      ManualTasks.
+   * @returns             A Promise, which resolves with the retrieved ManualTasks,
+   *                      or rejects an error, in case the request failed.
+   *                      This can happen, if the Correlation was not found,
+   *                      or the user is not authorized to see the it.
```
Comment (SteffenKn) [inline typescript/src/apis/imanual_task_consumer_api.ts]:
```suggestion
   *                      or the user is not authorized to see it.
```

File: typescript/src/apis/inotification_consumer_api.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+  /**
+   * Executes a callback when a process with a given ProcessModelId was started.
+   *
+   * @async
+   * @param identity       The requesting users identity.
+   * @param callback       The callback that will be executed when a new ProcessInstance
+   *                       was started. The message passed to the callback
+   *                       contains further information about the started process.
+   * @param processModelId Id of the ProcessModel for which created instance a
```
Comment (SteffenKn) [inline typescript/src/apis/inotification_consumer_api.ts]:
```suggestion
   * @param processModelId ID of the ProcessModel for which created instance a
```

File: typescript/src/apis/inotification_consumer_api.ts [hunk 6]
```diff
@@ -0,0 +1,145 @@
+import {IIdentity} from '@essential-projects/iam_contracts';
+
+import {Messages} from '../messages/index';
+
+/**
+ * The INotificationConsumerApi is used to manage Subscriptions for async
+ * notifications.
+ */
+export interface INotificationConsumerApi {
```
Comment (roblen45) [inline typescript/src/apis/inotification_consumer_api.ts]: Please move this methods to the specific interfaces:

onManualTaskWaiting to IManualTaskConsumerAPI
onUserTaskWaiting to IUserTaskConsumerAPI
etc.

**Comment by ElRaptorus** [comment]
@roblen45 @SteffenKn done

**Pull request merged**